both/controllers/taggedCourses.js

```diff
@@ -1,25 +0,0 @@
-TaggedCoursesController = AppController.extend({
-    waitOn: function() {
-        // Wait for the tagged resources to be available for the specific tag
-        return [
-            this.subscribe('taggedCourses', this.params.tag),
-            this.subscribe('images')
-        ];
-    },
-    data: function () {
-        return Courses.find();
```

Comment:
This set the data context to include all courses, which was why more courses were appearing than the current tag.